subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultConfigurationResolver.java

```diff
@@ -191,6 +202,82 @@ public void resolveArtifacts(ConfigurationInternal configuration, ResolverResult
         results.artifactsResolved(new DefaultResolvedConfiguration(result), result);
     }
 
+    @Override
+    public List<ArtifactRepository> getRepositories() {
+        List<ExposableRepository> exposableRepositories = CollectionUtils.collect(repositories, Transformers.cast(ExposableRepository.class));
+        List<ArtifactRepository> result = Lists.newArrayListWithExpectedSize(exposableRepositories.size());
+        for (ExposableRepository repository : exposableRepositories) {
+            ConfiguredModuleComponentRepository resolver = repository.createResolver();
+            result.add(ArtifactRepositoryImpl.from(
+                resolver.getId(),
+                getTypeOf(repository),
+                resolver.getName(),
+                repository.getProperties()
+            ));
+        }
+        return result;
+    }
+
+    private static class ArtifactRepositoryImpl implements ArtifactRepository {
+
+        private final String repositoryId;
+        private final String type;
+        private final String name;
+        private final Map<String, ?> properties;
+
+        private ArtifactRepositoryImpl(String repositoryId, RepositoryType type, String name, Map<String, ?> properties) {
+            this.repositoryId = repositoryId;
+            this.type = type.displayName;
+            this.name = name;
+            this.properties = ImmutableMap.copyOf(properties);
+        }
+
+        private static ArtifactRepositoryImpl from(String repositoryId, RepositoryType type, String name, Map<RepositoryPropertyType, ?> properties) {
+            Map<String, Object> props = new HashMap<String, Object>(properties.size());
+            for (Map.Entry<RepositoryPropertyType, ?> entry : properties.entrySet()) {
+                props.put(entry.getKey().displayName, entry.getValue());
+            }
+            return new ArtifactRepositoryImpl(
+                repositoryId,
+                type,
+                name,
+                props
+            );
+        }
+
+        @Override
+        public String getRepositoryId() {
+            return repositoryId;
+        }
+
+        @Override
+        public String getType() {
+            return type;
+        }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public Map<String, ?> getProperties() {
+            return properties;
+        }
+    }
+
+    private RepositoryType getTypeOf(ExposableRepository repository) {
```

Comment:
This should be on ExposableRepository, with an enum scheme like the property names.

Reply:
:x: +1

Reply:
done